### ISSUE TITLE
Tilde (~) is not expanded in LTeX path settings

### ISSUE THREAD

**Issue by Lewous** [valentjn/vscode-ltex#99]
Hi, 
The `LTeX` package was installed from .vsix file released on github. So, both `jdk 11.0` and `ltex-ls 6.2.0` is well prepared (and I've check manually). But a "Java failed" occurs when I try to set `ltex.ltex-ls.path`.

I think it has some relation with issue [#75](https://github.com/valentjn/vscode-ltex/issues/75). But since you've mentioned it has been fixed...

When the `ltex.ltex-ls.path` is not set (regardless of `ltex.java.path`), the `LTeX` works well. But when the `ltex.ltex-ls.path` is set to a local path `"~/my_git/ltex-ls-6.2.0"` , `Ltex` tell me "Test failed.". Here is the `OUTPUT` of `LTeX Language Client`:

```
2020-08-20T05:03:44.269Z Info: Searching for Java in '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib'...
2020-08-20T05:03:44.270Z Info: Java found in '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/jdk-11.0.8+10-jre/Contents/Home'.
2020-08-20T05:03:44.270Z Info: Using ltex-ls from '~/my_git/ltex-ls-6.2.0'.
2020-08-20T05:03:44.270Z Info: Using Java from '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/jdk-11.0.8+10-jre/Contents/Home'.
2020-08-20T05:03:44.270Z Info: Testing ltex-ls...
2020-08-20T05:03:44.270Z Info:   Command: "~/my_git/ltex-ls-6.2.0/bin/ltex-ls"
2020-08-20T05:03:44.270Z Info:   Arguments: ["--version"]
2020-08-20T05:03:44.270Z Info:   env['JAVA_HOME']: "/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/jdk-11.0.8+10-jre/Contents/Home"
2020-08-20T05:03:44.270Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-20T05:03:44.276Z Info: Test failed.
2020-08-20T05:03:44.277Z Info: Exit code of ltex-ls: null
2020-08-20T05:03:44.277Z Info: stdout of ltex-ls:
2020-08-20T05:03:44.277Z Info: 
2020-08-20T05:03:44.277Z Info: stderr of ltex-ls:
2020-08-20T05:03:44.277Z Info: 
2020-08-20T05:03:44.277Z Error: The download/extraction/run of Java failed!
2020-08-20T05:03:44.277Z Info: You might want to try offline installation, see $https://valentjn.github.io/vscode-ltex/docs/installation-how-to-use.html#offline-installation.

```

Then I tried this in my terminal:
```
lewousnoair:~ lewous_ttx$ cd my_git/ltex-ls-6.2.0/bin
lewousnoair:bin lewous_ttx$ ls 
ltex-ls		ltex-ls.bat
lewousnoair:bin lewous_ttx$ ./ltex-ls --version
{
  "ltex-ls": "6.2.0",
  "java": "14.0.2"
}
```

And when I set  `ltex.ltex-ls.path` to `"~/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0"`, this also happens:

```
2020-08-20T05:30:24.597Z Info: Searching for Java in '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib'...
2020-08-20T05:30:24.597Z Info: Java found in '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/jdk-11.0.8+10-jre/Contents/Home'.
2020-08-20T05:30:24.597Z Info: Using ltex-ls from '~/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0'.
2020-08-20T05:30:24.597Z Info: Using Java from '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/jdk-11.0.8+10-jre/Contents/Home'.
2020-08-20T05:30:24.597Z Info: Testing ltex-ls...
2020-08-20T05:30:24.597Z Info:   Command: "~/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0/bin/ltex-ls"
2020-08-20T05:30:24.597Z Info:   Arguments: ["--version"]
2020-08-20T05:30:24.597Z Info:   env['JAVA_HOME']: "/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/jdk-11.0.8+10-jre/Contents/Home"
2020-08-20T05:30:24.597Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-20T05:30:24.605Z Info: Test failed.
2020-08-20T05:30:24.605Z Info: Exit code of ltex-ls: null
2020-08-20T05:30:24.605Z Info: stdout of ltex-ls:
2020-08-20T05:30:24.605Z Info: 
2020-08-20T05:30:24.605Z Info: stderr of ltex-ls:
2020-08-20T05:30:24.605Z Info: 
2020-08-20T05:30:24.605Z Error: The download/extraction/run of Java failed!
2020-08-20T05:30:24.605Z Info: You might want to try offline installation, see $https://valentjn.github.io/vscode-ltex/docs/installation-how-to-use.html#offline-installation.
```

Here is the `OUTPUT` of `LTeX Language Client` when `ltex.ltex-ls.path` is not set (I set `ltex.java.path` to another local path this time):
```
2020-08-20T05:38:35.072Z Info: ltex.ltex-ls.path not set.
2020-08-20T05:38:35.072Z Info: Searching for ltex-ls in '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib'...
2020-08-20T05:38:35.074Z Info: ltex-ls found in '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0'.
2020-08-20T05:38:35.074Z Info: 
2020-08-20T05:38:35.074Z Info: ltex.java.path set to '/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/'.
2020-08-20T05:38:35.074Z Info: Using ltex-ls from '/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0'.
2020-08-20T05:38:35.074Z Info: Using Java from '/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/'.
2020-08-20T05:38:35.076Z Info: Testing ltex-ls...
2020-08-20T05:38:35.076Z Info:   Command: "/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0/bin/ltex-ls"
2020-08-20T05:38:35.076Z Info:   Arguments: ["--version"]
2020-08-20T05:38:35.076Z Info:   env['JAVA_HOME']: "/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/"
2020-08-20T05:38:35.076Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-20T05:38:37.878Z Info: Test successful!
2020-08-20T05:38:37.974Z Info: 
2020-08-20T05:38:37.983Z Info: Starting ltex-ls...
2020-08-20T05:38:37.983Z Info:   Command: "/Users/lewous_ttx/.vscode/extensions/valentjn.vscode-ltex-6.2.0/lib/ltex-ls-6.2.0/bin/ltex-ls"
2020-08-20T05:38:37.983Z Info:   Arguments: []
2020-08-20T05:38:37.983Z Info:   env['JAVA_HOME']: "/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home/"
2020-08-20T05:38:37.983Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-20T05:38:37.983Z Info: 
```

my VSCode version is :
```
Version: 1.48.0
Commit: db40434f562994116e5b21c24015a2e40b2504e6
Date: 2020-08-13T08:03:27.059Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 18.7.0
```

and my OS is : 
```
macOS Mojave
version 10.14.8
```

**Comment by Lewous** [comment]
This occurs because  `ltex.ltex-ls.path` do not understand the symbol `"~/"`. After the `"~/"` is replaced with full path, the `LTeX` works well.

**Comment by valentjn** [comment]
Tilde expansion is something that the shell (e.g., [Bash](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html)) does. `~` usually doesn't work in (most) other contexts, so I wouldn't say it's a bug in LT<sub>E</sub>X. However, since it's convenient, and it can confuse people if it doesn't work, this seems like a good feature request. This will be implemented 👍 

VS Code also supports tilde in some places: microsoft/vscode#36792

**Comment by valentjn** [comment]
Fix released in 6.3.0.
